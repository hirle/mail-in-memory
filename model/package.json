{
  "version": "1.0.0",
  "license": "MIT",
  "name": "@mail-in-memory/model",
  "author": "hirle",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist",
    "src"
  ],
  "engines": {
    "node": ">=12"
  },
  "scripts": {
    "build": "rollup --config",
    "test": "jest"
  },
  "devDependencies": {
    "jest": "^26.6.3",
    "rollup": "^2.35.1",
    "@rollup/plugin-typescript": "^8.1.0",
    "tslib": "^2.0.3",
    "typescript": "^4.1.3"
  },
  "prettier": {
    "printWidth": 80,
    "semi": true,
    "singleQuote": true,
    "trailingComma": "es5"
  },
  "module": "dist/model.esm.js"
}
