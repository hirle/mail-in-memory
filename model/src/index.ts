
export { default as Mail, MailInterface } from './Mail';
export { SocketMessages as SocketMessages } from './SocketMessages';