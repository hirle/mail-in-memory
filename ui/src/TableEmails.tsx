import React from 'react';
import {Mail} from '@mail-in-memory/model';
import 'antd/dist/antd.css';
import { Table } from 'antd';

interface TableEmailsProps {
    mails: Mail[]
}

const TableEmails: React.FunctionComponent<TableEmailsProps> = (props: TableEmailsProps) => {
    const columns = [
        {
            title: 'From',
            dataIndex: 'fromAddress',
            key: 'fromAddress',
        },
        {
            title: 'To',
            dataIndex: 'toAddress',
            key: 'toAddress',
        },
        {
            title: 'Subject',
            dataIndex: 'subject',
            key: 'subject',
        },
        {
            title: 'Body',
            dataIndex: 'body',
            key: 'body',
        },
        {
            title: 'TimeStamp',
            dataIndex: 'mailTimestamp',
            key: 'mailTimestamp',
            render: (date:Date) => date.toLocaleString(),
        }
    ];
    
    return <Table
        dataSource={props.mails}
        columns={columns}
        pagination ={{hideOnSinglePage:true}}
        rowKey='messageId'
        />;
}

export default TableEmails