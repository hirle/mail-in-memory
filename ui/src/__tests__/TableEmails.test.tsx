// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import {Mail} from '@mail-in-memory/model';
import TableEmails from '../TableEmails';

test('Link changes the class when hovered', () => {
    const testMails: Mail[] = [
        Mail.create( {
            fromAddress: 'foo@email.org',
            toAddress: 'jest@test.mail-in-memory',
            subject: 'Cool',
            body: 'With jest, unit testing is so cool',
            mailTimestamp: Date.parse('2021-02-21T18:13:00Z')
        }),
        Mail.create( {
            fromAddress: 'bar@email.org',
            toAddress: 'jest@test.mail-in-memory',
            subject: 'Great',
            body: 'Jest is really great',
            mailTimestamp: Date.parse('2021-02-21T18:13:10Z')
        })
    ];

  const component = renderer.create(
    <TableEmails 
        mails={testMails}
     />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});