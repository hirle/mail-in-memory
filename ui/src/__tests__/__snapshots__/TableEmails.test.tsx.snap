// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup />
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    From
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    To
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Subject
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Body
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    TimeStamp
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={1613931180000}
                  onClick={[Function]}
                  style={Object {}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    foo@email.org
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    jest@test.mail-in-memory
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Cool
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    With jest, unit testing is so cool
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    2021-2-21 19:13:00
                  </td>
                </tr>
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={1613931190000}
                  onClick={[Function]}
                  style={Object {}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    bar@email.org
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    jest@test.mail-in-memory
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Great
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Jest is really great
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    2021-2-21 19:13:10
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
